### ISSUE TITLE
Error in cov2cor

### ISSUE THREAD

**Issue by jindalk** [GabrielHoffman/dreamlet#27]
Hi, I ran the `dreamlet` function on my `dreamletProcessedData` object and it returned the following error: `Error in cov2cor(C) : 'V' is not a square numeric matrix`

The exact error seems to be outside of core dreamlet functions, but is there any guidance on how to go about debugging this?

Traceback attached:

```
16: stop("'V' is not a square numeric matrix")
15: cov2cor(C)
14: .local(object, ...)
13: classifyTestsF(obj, df = obj$df.total, fstat.only = TRUE)
12: classifyTestsF(obj, df = obj$df.total, fstat.only = TRUE)
11: `[.MArrayLM2`(out, , realcoef)
10: out[, realcoef]
9: is.data.frame(x)
8: rowMeans(out[, realcoef]$df.total)
7: .standard_transform(fit_eb, sigma = sqrt(fit_eb$s2.post))
6: eBayes(fit, robust = robust, trend = !isCounts)
5: eBayes(fit, robust = robust, trend = !isCounts)
4: FUN(X[[i]], ...)
3: lapply(assays, function(k) {
       if (!quiet) 
           message("  ", k, "...", appendLF = FALSE)
       startTime <- Sys.time()
       geneExpr <- assay(x, k)
       ids <- intersect(colnames(geneExpr), rownames(data_constant))
       geneExpr <- geneExpr[, ids, drop = FALSE]
       data2 <- merge_metadata(data_constant[ids, , drop = FALSE], 
           metadata(x), k, x@by)
       form_mod <- removeConstantTerms(formula, data2)
       form_mod <- dropRedundantTerms(form_mod, data2)
       errorArray <- NULL
       if (length(all.vars(form_mod)) > 0 & isFullRank(form_mod, 
           data2)) {
           if (!is.null(contrasts)) {
               L <- makeContrastsDream(form_mod, data2, contrasts = contrasts, 
                   nullOnError = TRUE)
           }
           else {
               L <- NULL
    ...
2: dreamlet(res_proc, formula_use, BPPARAM = SnowParam(8))
1: dreamlet(res_proc, formula_use, BPPARAM = SnowParam(8))
```

**Comment by GabrielHoffman** [comment]
1) can you show `sessionInfo()`
2) can you give a minimal reproducible example

`dreamlet` should catch that error and handle it gracefully.

**Comment by jindalk** [comment]
I'll work on obtaining a shareable reproducible example. Meanwhile, here is the session info:

```
> sessionInfo()
R version 4.3.0 (2023-04-21)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: FlexiBLAS OPENBLAS;  LAPACK version 3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: America/New_York
tzcode source: system (glibc)

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] plotly_4.10.4               shiny_1.7.4                 DESeq2_1.42.1               circlize_0.4.16            
 [5] ComplexHeatmap_2.18.0       Seurat_5.1.0                SeuratObject_5.0.2          sp_2.1-4                   
 [9] ggrepel_0.9.5               qvalue_2.34.0               viridis_0.6.5               viridisLite_0.4.2          
[13] tidyr_1.3.1                 Matrix_1.6-5                scater_1.30.1               scuttle_1.12.0             
[17] zenith_1.4.2                muscat_1.16.0               zellkonverter_1.12.1        MOFA2_1.12.1               
[21] data.table_1.15.4           pathview_1.42.0             dplyr_1.1.4                 purrr_1.0.2                
[25] org.Hs.eg.db_3.18.0         AnnotationDbi_1.64.1        DOSE_3.28.2                 dreamlet_1.0.3             
[29] SingleCellExperiment_1.24.0 SummarizedExperiment_1.32.0 Biobase_2.60.0              GenomicRanges_1.52.1       
[33] GenomeInfoDb_1.36.4         IRanges_2.34.1              S4Vectors_0.38.2            BiocGenerics_0.46.0        
[37] MatrixGenerics_1.14.0       matrixStats_1.3.0           variancePartition_1.32.5    BiocParallel_1.36.0        
[41] limma_3.58.1                ggplot2_3.5.1               enrichplot_1.22.0           clusterProfiler_4.10.1     

loaded via a namespace (and not attached):
  [1] GSEABase_1.62.0           progress_1.2.3            urlchecker_1.0.1          goftest_1.2-3            
  [5] Biostrings_2.68.1         HDF5Array_1.30.1          vctrs_0.6.5               spatstat.random_3.2-3    
  [9] digest_0.6.35             png_0.1-8                 corpcor_1.6.10            shape_1.4.6.1            
 [13] mixsqp_0.3-54             deldir_2.0-4              corrplot_0.92             parallelly_1.37.1        
 [17] MASS_7.3-58.1             reshape2_1.4.4            SQUAREM_2021.1            httpuv_1.6.8             
 [21] foreach_1.5.2             withr_2.5.0               ggfun_0.1.6               ellipsis_0.3.2           
 [25] survival_3.7-0            memoise_2.0.1             ggbeeswarm_0.7.2          gson_0.1.0               
 [29] profvis_0.3.8             systemfonts_1.1.0         ragg_1.3.2                zoo_1.8-12               
 [33] tidytree_0.4.6            GlobalOptions_0.1.2       gtools_3.9.5              KEGGgraph_1.60.0         
 [37] pbapply_1.7-2             prettyunits_1.2.0         promises_1.2.0.1          KEGGREST_1.40.1          
 [41] httr_1.4.7                fitdistrplus_1.1-11       globals_0.16.3            rhdf5filters_1.14.1      
 [45] ps_1.7.6                  ashr_2.2-63               rhdf5_2.46.1              rstudioapi_0.16.0        
 [49] miniUI_0.1.1.1            generics_0.1.3            processx_3.8.4            dir.expiry_1.10.0        
 [53] curl_5.2.1                babelgene_22.9            zlibbioc_1.46.0           ScaledMatrix_1.10.0      
 [57] ggraph_2.2.1              polyclip_1.10-6           RcppZiggurat_0.1.6        GenomeInfoDbData_1.2.10  
 [61] SparseArray_1.2.4         desc_1.4.3                xtable_1.8-4              stringr_1.5.0            
 [65] doParallel_1.0.17         S4Arrays_1.2.1            Rfast_2.1.0               hms_1.1.3                
 [69] irlba_2.3.5.1             colorspace_2.1-0          filelock_1.0.3            ROCR_1.0-11              
 [73] spatstat.data_3.0-4       reticulate_1.37.0         lmtest_0.9-40             magrittr_2.0.3           
 [77] Rgraphviz_2.44.0          later_1.3.0               ggtree_3.10.1             lattice_0.22-6           
 [81] spatstat.geom_3.2-9       future.apply_1.11.2       scattermore_1.2           XML_3.99-0.16.1          
 [85] shadowtext_0.1.4          cowplot_1.1.3             RcppAnnoy_0.0.22          pillar_1.9.0             
 [89] nlme_3.1-165              iterators_1.0.14          caTools_1.18.2            compiler_4.3.0           
 [93] beachmat_2.18.1           RSpectra_0.16-1           stringi_1.7.12            rmeta_3.0                
 [97] devtools_2.4.5            tensor_1.5                minqa_1.2.7               plyr_1.8.9               
[101] msigdbr_7.5.1             crayon_1.5.2              abind_1.4-5               truncnorm_1.0-9          
[105] blme_1.0-5                metadat_1.2-0             gridGraphics_0.5-1        locfit_1.5-9.9           
[109] graphlayouts_1.1.1        bit_4.0.5                 mathjaxr_1.6-0            fastmatch_1.1-4          
[113] textshaping_0.4.0         codetools_0.2-20          BiocSingular_1.18.0       GetoptLong_1.0.5         
[117] mime_0.12                 remaCor_0.0.18            splines_4.3.0             fastDummies_1.7.3        
[121] Rcpp_1.0.10               basilisk_1.14.3           sparseMatrixStats_1.14.0  HDO.db_0.99.1            
[125] EnrichmentBrowser_2.32.0  blob_1.2.4                utf8_1.2.4                clue_0.3-65              
[129] lme4_1.1-35.3             fs_1.6.2                  listenv_0.9.1             DelayedMatrixStats_1.24.0
[133] Rdpack_2.6                pkgbuild_1.4.4            ggplotify_0.1.2           tibble_3.2.1             
[137] callr_3.7.6               fANCOVA_0.6-1             tweenr_2.0.3              pkgconfig_2.0.3          
[141] pheatmap_1.0.12           tools_4.3.0               cachem_1.0.7              RhpcBLASctl_0.23-42      
[145] rbibutils_2.2.16          RSQLite_2.3.7             DBI_1.2.3                 numDeriv_2016.8-1.1      
[149] fastmap_1.1.1             scales_1.3.0              usethis_2.2.3             ica_1.0-3                
[153] broom_1.0.6               patchwork_1.2.0           dotCall64_1.1-1           graph_1.78.0             
[157] RANN_2.6.1                snow_0.4-4                farver_2.1.2              aod_1.3.3                
[161] tidygraph_1.3.1           scatterpie_0.2.4          mgcv_1.9-1                cli_3.6.1                
[165] leiden_0.4.3.1            lifecycle_1.0.3           mashr_0.2.79              uwot_0.2.2               
[169] glmmTMB_1.1.9             mvtnorm_1.2-5             sessioninfo_1.2.2         backports_1.5.0          
[173] annotate_1.78.0           gtable_0.3.5              rjson_0.2.21              ggridges_0.5.6           
[177] metafor_4.6-0             progressr_0.14.0          parallel_4.3.0            ape_5.8                  
[181] jsonlite_1.8.8            edgeR_3.42.4              RcppHNSW_0.6.0            bitops_1.0-7             
[185] bit64_4.0.5               assertthat_0.2.1          Rtsne_0.17                yulab.utils_0.1.7        
[189] spatstat.utils_3.0-4      BiocNeighbors_1.20.2      RcppParallel_5.1.7        GOSemSim_2.28.1          
[193] pbkrtest_0.5.2            lazyeval_0.2.2            htmltools_0.5.8.1         GO.db_3.17.0             
[197] sctransform_0.4.1         basilisk.utils_1.14.1     glue_1.6.2                spam_2.10-0              
[201] XVector_0.40.0            RCurl_1.98-1.14           treeio_1.26.0             gridExtra_2.3            
[205] EnvStats_2.8.1            boot_1.3-30               igraph_2.0.3              TMB_1.9.11               
[209] invgamma_1.1              R6_2.5.1                  gplots_3.1.3.1            labeling_0.4.3           
[213] forcats_1.0.0             cluster_2.1.6             pkgload_1.3.4             Rhdf5lib_1.24.2          
[217] aplot_0.2.3               nloptr_2.0.3              DelayedArray_0.28.0       tidyselect_1.2.1         
[221] vipor_0.4.7               ggforce_0.4.2             future_1.33.2             rsvd_1.0.5               
[225] munsell_0.5.1             KernSmooth_2.23-24        htmlwidgets_1.6.4         fgsea_1.28.0             
[229] RColorBrewer_1.1-3        spatstat.sparse_3.0-3     rlang_1.1.4               spatstat.explore_3.2-7   
[233] remotes_2.5.0             lmerTest_3.1-3            fansi_1.0.6               beeswarm_0.4.0   
```

**Comment by GabrielHoffman** [comment]
I see you have `dreamlet_1.0.3`.  Try a more recent version

**Comment by jindalk** [comment]
I tried installing the latest version, but it requires limma >= 3.62.1 which isn't available for R 4.3 Is there a new-ish version of dreamlet that could work with R4.3? If not, I'll work on getting R4.4 set up.

Thanks

**Comment by GabrielHoffman** [comment]
Try `devtools::install_github("DiseaseNeuroGenomics/dreamlet@785edd6d7f299df3fadbd59fd7cc9077080317a7"`)
